ftlib/consensus/gossip/__init__.py

```diff
@@ -0,0 +1,110 @@
+__version__ = '0.0.1'
+
+import time
+import logging
+from ctypes import *
+from ..basic import BasicConsensus
+from ..consensus_status import ConsensusStatus
+
+
+class Gossip(BasicConsensus):
+    class member_list(Structure):
+        _fields_ = [('addrs', (c_char_p * 1024)), ('size', c_int)]
+
+        def __len__(self):
+            return self.size
+
+        def __getitem__(self, idx, codec='utf-8'):
+            if idx >= self.__len__():
+                raise IndexError(
+                    "list index out of range: trying to get {idx} while only has {item_num}"
+                    .format(idx=idx, item_num=self.__len__()))
+            return self.addrs[idx].decode(codec)
+
+        def __eq__(self, val):
+            val_set = set([val[i] for i in range(len(val))])
+            for i in range(len(self)):
+                if self[i] not in val_set:
+                    return False
+                val_set.remove(self[i])
+            return len(val_set) == 0
+
+        def __contains__(self, item, codec='utf-8'):
+            for idx in range(self.size):
+                if self.addrs[idx].decode(codec) == item:
+                    return True
+            return False
+
+    def __init__(self,
+                 ftlib,
+                 known_addr_list,
+                 so_path='memberlist.so',
+                 log_path='/tmp/memberlist.log'):
+        super(Gossip, self).__init__()
+
+        self._ftlib = ftlib
+
+        self._lib = cdll.LoadLibrary(so_path)
```

Comment:
Where does the cdll come from? Maybe from `from ctypes import *`?

Reply:
right, I extract all used definition from ctypes.